lib/models/user.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+require 'active_record'
+
+# User model
+class User < ActiveRecord::Base
+  self.primary_key = :id
+  validates :login, :email, presence: true
+
+  # after_commit :create_master_key
+
+  # private
+
+  def create_master_key
+    update(master_key: Digest::SHA256.hexdigest(id.to_s))
```

Comment:
I don't think this will work unless the object is already persisted. Maybe a class method? `User.update_master_key(user_id)`.